### DISCUSSION TITLE
Open source projects using sqlc

### DISCUSSION THREAD

**Discussion by joeshaw** [sqlc-dev/sqlc#834] [Issue Triage]
Hi,

Are you aware of any open source projects that are using sqlc?  While I appreciate the examples included in the sqlc repo, I'd also like to see some larger examples of using it in action.  With many Go packages it's fairly easy to search on pkg.go.dev, but with sqlc it's hard because it's not imported by projects that use it.  Any pointers you can provide would be appreciated!

Thanks,
Joe

**Comment by kevinburke1** [comment]
https://github.com/kevinburke/rickover uses it

**Reply by kevinburke1** [reply]
(note; the code there slightly more awkward than it should be because we ported the code from the old "ORM" to sqlc)

**Reply by HoangNguyen689** [reply]
@kevinburke1 
Can I ask Why you use the models layer than use the newmodels directly?

**Comment by kevinburke1** [comment]
We're also using sqlc pretty heavily to build meter.com, but unfortunately the code is private. We are hiring though.

**Comment by kyleconroy** [comment]
> Are you aware of any open source projects that are using sqlc?

You're right that you can't use pkg.go.dev to find projects using sqlc. However, you can use GitHub code search. I found [235 repositories](https://github.com/search?l=&p=1&q=filename%3Asqlc.json+filename%3Asqlc.yaml&type=Code) containing a sqlc.json / sqlc.yaml file.

That said, as Kevin mentioned, there are many organizations using sqlc. Three of the biggest sponsors are organizations (https://ngrok.com, https://meter.com, http://www.getweave.com) that use it heavily. I'd be happy to introduce you to them, just drop me an email at kyle@conroy.org.

**Reply by joeshaw** [reply]
I tried the GitHub search route and while it's helpful there's a lot of noise in there.  What I was hoping for more was a project (or two, or three) that you felt really showcased sqlc and could act as a good guide for best practices in using it within a larger project.

I'm happy to see it's getting uptake in organizations, but that's not what I'm personally looking for right now.  I'm very grateful the project exists... writing a HugSQL-like for Go was something that was on my aspirational to-do list for _years_ but never materialized.  Thank you.

**Reply by kyleconroy** [reply]
I have a few projects that meet that criteria, except they are not open source. I'd be happy to give you access to them if you'd just like to look around.

**Reply by joeshaw** [reply]
Thanks! I’ll send you an email.
